src/worker.cc

```diff
@@ -555,6 +555,20 @@ void from_json(rapidjson::Document& doc, odin::DirectionsOptions& options) {
     }
   }
 
+  // Set the output precision for shape/geometry (polyline encoding). Defaults to polyline6
+  // TODO - is this just for OSRM compatibility?
+  options.set_shape_format(odin::polyline6);
+  auto shape_format = rapidjson::get_optional<std::string>(doc, "/shape_format");
+  if (shape_format) {
+    if (*shape_format == "polyline6") {
+      options.set_shape_format(odin::polyline6);
+    } else if (*shape_format == "polyline5") {
+      options.set_shape_format(odin::polyline5);
+    } else {
+      ; // TODO - error or just ignore?
```

Comment:
i think we should set it to default of `polyline6` and silently continue

Reply:
I think an error would be better here.  The IETF argues that protocol deviations should [fail fast and hard](https://tools.ietf.org/html/draft-thomson-postel-was-wrong-00#section-4.1) - and the API interface here is the protocol between client and Valhalla.  Aiming for maximum strictness gives us more confidence in the compatibility consequences of further changes here in the future (e.g. adding `geojson` as a supported type).

Reply:
fair enough! we can add an error code

Reply:
done